tf2_ros/src/static_transform_broadcaster_program.cpp

```diff
@@ -64,6 +66,10 @@ int main(int argc, char ** argv)
   // TODO(clalancette): Anonymize the node name like it is in ROS1.
   auto node = rclcpp::node::Node::make_shared("static_transform_publisher");
 
+  rclcpp::TimeSource ts(node);
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
+  ts.attachClock(clock);
```

Comment:
Maybe it's just a convenience function, but an API like this would be much nicer imo:

```c++
auto node = rclcpp::node::Node::make_shared("static_transform_publisher");

rclcpp::Clock::SharedPtr clock = node->get_ros_time_clock();
```

Or even just this:

```c++
msg.head.stamp = node->now();
```

At the very least, why do you need to pass `RCL_ROS_TIME` to the clock? Is that really useful information for the user to have to manually specify? Is that important in this case in particular or is that just the standard boilerplate of this API?

Reply:
That's targeted for the next stage of development.

Reply:
> That's targeted for the next stage of development.

Which part:

- `rclcpp::Node::get_ros_time_clock()`
- `rclcpp::Node::now()`
- not having to pass `RCL_ROS_TIME`?

Reply:
The Node API is targeted next. I'm expecting to have an internal clock that can pass through both those calls.

The default currently is RCL_SYSTEM_TIME though that can be changed. This is a lower level API that I would expect most people won't use once the node API is implemented. I can change the default to RCL_ROS_TIME instead.